st2tests/st2tests/base.py

```diff
@@ -402,6 +403,81 @@ def assertProcessExited(self, proc):
             self.fail('Process with pid "%s" is still running' % (proc.pid))
 
 
+class BaseSensorTestCase(TestCase):
+    """
+    Base class for sensor tests.
+
+    This class provides some utility methods for verifying that a trigger has
+    been dispatched, etc.
+    """
+
+    sensor_cls = None
+
+    def setUp(self):
```

Comment:
Yes - it's intentional. This way it's called before every test method so all the internal state on the sensor service is cleared between each tests.